contrib/pyln-proto/pyln/proto/onion.py

```diff
@@ -242,6 +256,238 @@ class SignatureField(TlvField):
     pass
 
 
+VERSION_SIZE = 1
+REALM_SIZE = 1
+HMAC_SIZE = 32
+PUBKEY_SIZE = 33
+ROUTING_INFO_SIZE = 1300
+TOTAL_PACKET_SIZE = VERSION_SIZE + PUBKEY_SIZE + HMAC_SIZE + ROUTING_INFO_SIZE
+
+
+class RoutingOnion(object):
+    def __init__(
+            self, version: int,
+            ephemeralkey: PublicKey,
+            payloads: bytes,
+            hmac: bytes
+    ):
+        assert(len(payloads) == ROUTING_INFO_SIZE)
+        self.version = version
+        self.payloads = payloads
+        self.ephemeralkey = ephemeralkey
+        self.hmac = hmac
+
+    @classmethod
+    def from_bin(cls, b: bytes):
+        if len(b) != TOTAL_PACKET_SIZE:
+            raise ValueError(
+                "Encoded binary RoutingOnion size mismatch: {} != {}".format(
+                    len(b), TOTAL_PACKET_SIZE
+                )
+            )
+
+        version = int(b[0])
+        ephemeralkey = PublicKey(b[1:34])
+        payloads = b[34:1334]
+        hmac = b[1334:]
+
+        assert(len(payloads) == ROUTING_INFO_SIZE
+               and len(hmac) == HMAC_SIZE)
+        return cls(version=version, ephemeralkey=ephemeralkey,
+                   payloads=payloads, hmac=hmac)
+
+    @classmethod
+    def from_hex(cls, s: str):
+        return cls.from_bin(unhexlify(s))
```

Comment:
we should really wean off unhexlify and hexlify in favor of bytes.fromhex() and bytes.hex()...